amy/extrequests/utils.py

```diff
@@ -72,3 +64,15 @@ def member_code_valid_training(
         raise MemberCodeValidationError("Membership has no training seats remaining.")
 
     return True
+
+
+def get_membership_from_code_if_exists(code: str) -> Membership | None:
```

Comment:
Naming suggestion: `get_membership_by_code_or_none`, as `or_none` automatically suggest what is returned if membership is not found.

On the other hand, my suggestion may be flawed by suggesting fetching the membership "by code or by None".